### PR TITLE
Better error message for pattern matching

### PR DESCRIPTION
**Pull request by forki**
For the following error:

image

we now have

image

and for

image

we now show

image

### CODE REVIEW

**Comment by forki** [comment]
image

**Comment by forki** [comment]
Agent went offline during the build - argh!!!!

**Comment by cartermp** [comment]
@mmitche Would you know what's going on with Jenkins going offline?  It seems to happen every other build or so.

File: src/fsharp/FSComp.txt [hunk 1]
```diff
@@ -18,6 +18,8 @@ undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined
 replaceWithSuggestion,"Replace with '%s'"
 missingElseBranch,"The 'if' expression is missing an 'else' branch. The 'then' branch has type '%s'. Because 'if' is an expression, and not a statement, add an 'else' branch which returns a value of the same type."
 elseBranchHasWrongType,"All branches of an 'if' expression must return the same type. This expression was expected to have type '%s' but here has type '%s'."
+followingPatternMatchClauseHasWrongType,"All branches of a pattern match expression must return the same type. This expression was expected to have type '%s' but here has type '%s'."
```
Comment (saul) [inline src/fsharp/FSComp.txt]: "must return the same type" -> "must be of the same type"

Comment (KevinRansom) [inline src/fsharp/FSComp.txt]: @forki --- not wishing to bikeshed, but I agree with @saul.

**Comment by mmitche** [comment]
@cartermp Actually it's not Jenkins in this case.  There is some kind of bugcheck happening occasionally on Win2016 which is disconnecting machines randomly.

**Comment by mmitche** [comment]
@cartermp I think the right thing to do here is to run new updates and hope it's been fixed.  However, I don't think this happened till recently.

File: src/fsharp/FSComp.txt (hunk 1 above)
Comment (forki) [inline src/fsharp/FSComp.txt]: Yep. I agree. 
Will also fix the other messages that we already have in there that say "return"

**Comment by forki** [comment]
> Win2016 which is disconnecting machines randomly

What? Could you please not do this on (semi-) public build servers? This is a nightmare to work with.

**Comment by forki** [comment]
Error message fixed

**Comment by forki** [comment]
@KevinRansom what did you do here? merged master into matchclauses branch? Did you inted to do the inverse?

**Comment by forki** [comment]
also please note that bec651c was green, but after your merge a already fixed error got introduced. please merge bec651c to master. thx

**Comment by 0x53A** [comment]
@forki It looks like there are new tests on master, where the also message needs to be updated:

image

**Comment by forki** [comment]
ok rebased again

**Comment by forki** [comment]
done. cc @KevinRansom

**Pull request merged**